src/Microsoft.AspNetCore.Sockets.Common/TransportType.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    [Flags]
+    public enum TransportType
+    {
+        WebSockets = 1,
+        ServerSentEvents = 2,
+        LongPolling = 4,
+        All = WebSockets | ServerSentEvents | LongPolling
```

Comment:
Suppose we use this enum to allow the user to pick the transport they want to use. What it would mean if the user passes TransportType.All? Can .All be defined outside this enum as a const or not at all? We pollute a public type and then use .All only in one place.